app/Http/Controllers/SubmissionController.php

```diff
@@ -129,6 +131,40 @@ private function submitProcess(): Response
             abort(Response::HTTP_NOT_FOUND, 'The requested project does not exist.');
         }
 
+        // Figure out what type of XML file this is.
+        try {
+            $stored_filename  = "inbox/".$filename;
+            $xml_info = SubmissionUtils::get_xml_type(fopen(Storage::path($stored_filename), 'r'), $stored_filename);
+        } catch (CDashXMLValidationException $e) {
+            foreach ($e->getDecodedMessage() as $error) {
+                Log::error($error);
+            }
+            abort(Response::HTTP_NOT_FOUND, 'Unable to determine the Type of the submission file.');
+        }
+        $filehandle = $xml_info['file_handle'];
+        $handler_ref = $xml_info['xml_handler'];
+        $file = $xml_info['xml_type'];
+
+        // If validation is enabled and if this file has a corresponding schema, validate it
+        $handler = new $handler_ref($this->project);
```

Comment:
As we discussed in person, it might be better to call a static method instead of instantiating a handler each time.